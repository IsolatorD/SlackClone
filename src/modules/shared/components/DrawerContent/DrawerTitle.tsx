import React from "react";
import { Text, View } from "react-native";
import DawerTitleStyles from "./DrawerTitleStyles";

interface DrawerTitleProps {
  title: string;
  isDarkMode: boolean;
}

export default function DrawerTitle({ title, isDarkMode }: DrawerTitleProps) {
  return (
    <View
      testID="drawer-title-container"
      style={DawerTitleStyles.titleContainer}
    >
      <Text
        testID="drawer-title"
        allowFontScaling={false}
        style={[
          DawerTitleStyles.title,
          isDarkMode && DawerTitleStyles.titleDark
        ]}
      >
        {title}
      </Text>
    </View>
  )
}