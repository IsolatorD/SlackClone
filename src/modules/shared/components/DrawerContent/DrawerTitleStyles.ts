import { StyleSheet } from 'react-native'
import ThemeColors from '../../constants/theme'

export default StyleSheet.create({
  titleContainer: {
    paddingHorizontal: 10,
    paddingVertical: 15,
  },
  title: {
    fontSize: 18,
    fontWeight: "bold",
    color: ThemeColors.black
  },
  titleDark: {
    color: ThemeColors.white
  },
})