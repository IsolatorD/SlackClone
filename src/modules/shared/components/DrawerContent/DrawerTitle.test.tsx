import 'react-native';
import React from 'react';
import DrawerTitle from './DrawerTitle';
import { render } from '@testing-library/react-native';
// @ts-ignore
import { toHaveStyle, toHaveTextContent } from '@testing-library/jest-native';
import ThemeColors from '../../constants/theme';
import DrawerTitleStyles from './DrawerTitleStyles';

expect.extend({ toHaveStyle, toHaveTextContent });

// group: DrawerTitle
describe('DrawerTitle.tsx', () => {
  let component: any = null;
  beforeEach(() => {
    component = render(<DrawerTitle title='Espacios de trabajo' isDarkMode />);
  });

  it('Component render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('View component render correctly', () => {
    expect(component.getByTestId('drawer-title-container')).toBeDefined();
  })

  it('Text component render correctly', () => {
    expect(component.getByTestId('drawer-title')).toBeDefined();
  })

  it('View container component render styles correctly', () => {
    const titleComponent = component.getByTestId('drawer-title-container');
    expect(titleComponent).toHaveStyle(DrawerTitleStyles.titleContainer);
  })
  
  it('Component render initial title correctly', () => {
    const titleComponent = component.getByTestId('drawer-title');
    expect(titleComponent).toHaveTextContent('Espacios de trabajo');
  })

  it('Component render title with dark theme color by passing "isDarkMode" prop with value "true"', () => {
    const titleComponent = component.getByTestId('drawer-title');
    expect(titleComponent).toHaveStyle({
      ...DrawerTitleStyles.title,
      color: ThemeColors.white,
    });
  })

  it('Component render title with light theme color by passing "isDarkMode" prop with value "false"', () => {
    component.rerender(<DrawerTitle title='Espacios de trabajo' isDarkMode={false} />);
    const titleComponent = component.getByTestId('drawer-title');
    expect(titleComponent).toHaveStyle(DrawerTitleStyles.title);
  })

  it('Component render new title text correctly when "title" prop change', () => {
    component.rerender(<DrawerTitle title='Nuevo titulo' isDarkMode={false} />);
    const titleComponent = component.getByTestId('drawer-title');
    expect(titleComponent).toHaveTextContent('Nuevo titulo');
  })
});

