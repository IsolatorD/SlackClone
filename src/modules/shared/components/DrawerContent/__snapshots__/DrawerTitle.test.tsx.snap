// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerTitle.tsx Component render correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "paddingVertical": 15,
    }
  }
  testID="drawer-title-container"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 18,
          "fontWeight": "bold",
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
    testID="drawer-title"
  >
    Espacios de trabajo
  </Text>
</View>
`;
